internal/pkg/pct/pct_test.go

```diff
@@ -213,304 +305,39 @@ func TestGet(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := Get(tt.args.templateCache, tt.args.selectedTemplate)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("Get() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("Get() = %v, want %v", got, tt.want)
-			}
-		})
-	}
-}
-
-func Test_createTemplateFile(t *testing.T) {
-	type args struct {
-		info         DeployInfo
-		configFile   string
-		templateFile PuppetContentTemplateFileInfo
-		tmpl         PuppetContentTemplate
-	}
-
-	tmp := t.TempDir()
+			fs := afero.NewMemMapFs()
+			afs := &afero.Afero{Fs: fs}
+			iofs := &afero.IOFS{Fs: fs}
 
-	tests := []struct {
-		name    string
-		args    args
-		wantErr bool
-	}{
-		{
-			name: "",
-			args: args{
-				info: DeployInfo{
-					TargetName: "foobar",
-					PdkInfo: PDKInfo{
-						Version:   "0.1.0",
-						Commit:    "abc12345",
-						BuildDate: "2021/06/27",
-					},
-				},
-				configFile: "testdata/examples/good-project/pct.yml",
-				tmpl: PuppetContentTemplate{
-					Type:    "project",
-					Display: "Good Project",
-					URL:     "https://github.com/puppetlabs/pct-good-project",
-					Version: "0.1.0",
-					Id:      "good-project",
-				},
-				templateFile: PuppetContentTemplateFileInfo{
-					TemplatePath:   "testdata/examples/good-project/content/goodfile.txt.tmpl",
-					TargetFilePath: filepath.Join(tmp, "foo.txt"),
-					TargetDir:      tmp,
-					TargetFile:     "",
-					IsDirectory:    false,
-				},
-			},
-			wantErr: false,
-		},
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			if err := createTemplateFile(tt.args.info, tt.args.configFile, tt.args.templateFile, tt.args.tmpl); (err != nil) != tt.wantErr {
-				t.Errorf("createTemplateFile() error = %v, wantErr %v", err, tt.wantErr)
-			}
-			if _, err := os.Stat(tt.args.templateFile.TargetFilePath); err != nil {
-				t.Errorf("createTemplateFile() error = %v, wantErr %v", err, tt.wantErr)
+			if tt.args.setup {
+				// Create the template
+				templateDir := filepath.Join(tt.args.templateCache, tt.args.selectedTemplate)
+				contentDir := filepath.Join(templateDir, "content")
+				afs.MkdirAll(contentDir, 755) //nolint:errcheck
```

Comment:
Similar to my earlier comment:
```suggestion
				afs.MkdirAll(contentDir, 750)
```